### ISSUE TITLE
Spurious error `Duplicate identifier 'module'` reported in node_modules dependency

### ISSUE THREAD

**Issue by BurtHarris** [palantir/tslint#3042]
### Bug Report

- __TSLint version__: 5.5.0
- __TypeScript version__: 2.4.1
- __Running TSLint via__: VSCode 1.13.1 / TSLint addin version 0.16.0

#### TypeScript code being linted

https://github.com/hegemonic/requizzle/blob/master/lib/requizzle.js#L20 (only because it's a dependency of jsdoc, which I'm trying to use as a dev dependency.)  The relevant part of requizzle.js seems to consist of the following:

```ts
'use strict';

/** @module lib/requizzle */

var _ = require('underscore');
var loader = require('./loader');
var Module = require('module');
var path = require('path');

/**
 * Function that returns text to swizzle into the module.
 *
 * @typedef module:lib/requizzle~wrapperFunction
 * @type {function}
 * @param {string} targetPath - The path to the target module.
 * @param {string} parentModulePath - The path to the module that is requiring the target module.
 * @return {string} The text to insert before or after the module's source code.
 */

/**
 * Options for the wrappers that will be swizzled into the target module.
 *
 * @typedef module:lib/requizzle~options
 * @type {Object}
 * @property {Object=} options.extras - Functions that generate text to swizzle into the target
 * module.
 * @property {module:lib/requizzle~wrapperFunction} options.extras.after - Function that returns
 * text to insert after the module's source code.
 * @property {module:lib/requizzle~wrapperFunction} options.extras.before - Function that returns
 * text to insert before the module's source code.
 * @property {(Array.<string>|string)} options.requirePaths - Additional paths to search when
 * resolving module paths in the target module.
 */

function isNativeModule(targetPath, parentModule) {
```

with `tslint.json` configuration:

```json
{
    "extends": [
        "tslint-config-standard"
    ],
    "rules": {
        // producing conflicts...
        "ter-indent": false,
        "no-unused-variable": false

    }
}
```

#### Actual behavior

With a VSCode open to debug code in this file, VSCode treats JSDoc comments in this dependency as errors, w/ red squiggles & "problems" reported on the two lines starting with "@typedef"

#### Expected behavior

I really don't want dependencies linted by default.

**Comment by ajafff** [comment]
@BurtHarris that's no error from tslint, it comes from typescript. You probably see it as `[ts] Duplicate identifier 'module'`.
There's nothing tslint can do about this.

If it also happens when you run tslint CLI, just omit the `--type-check` flag

**Comment by BurtHarris** [comment]
Interesting, thank you.
